packages/hub/node-tests/routes/profiles-test.ts

```diff
@@ -0,0 +1,669 @@
+import { registry, setupHub, setupRegistry } from '../helpers/server';
+import { v4 as uuidv4 } from 'uuid';
+
+const stubNonce = 'abc:123';
+let stubAuthToken = 'def--456';
+let stubTimestamp = process.hrtime.bigint();
+
+class StubAuthenticationUtils {
+  generateNonce() {
+    return stubNonce;
+  }
+  buildAuthToken() {
+    return stubAuthToken;
+  }
+  extractVerifiedTimestamp(_nonce: string) {
+    return stubTimestamp;
+  }
+
+  validateAuthToken(encryptedAuthToken: string) {
+    return handleValidateAuthToken(encryptedAuthToken);
+  }
+}
+
+let stubUserAddress = '0x2f58630CA445Ab1a6DE2Bb9892AA2e1d60876C13';
+function handleValidateAuthToken(encryptedString: string) {
+  expect(encryptedString).to.equal('abc123--def456--ghi789');
+  return stubUserAddress;
+}
+
+describe('/api/profiles', function () {
+  describe('GET /api/profiles/validate-slug/:slug', function () {
+    this.beforeEach(function () {
+      registry(this).register('authentication-utils', StubAuthenticationUtils);
```

Comment:
Pre-existing problem - we don't check for worker jobs to update S3, which is the more important source of truth for web client + wallet most of the time.